### PR TITLE
chore: Remove obsolete legacy visualizations

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

Whilst working on https://github.com/apache/superset/pull/24671 I realize that some non-legacy charts (which are invoked via the `/api/v1/chart` RESTful API endpoint) are still present in `superset.viz` meaning the logic outlined in said PR for differentiating between legacy and non-legacy charts was incorrect.

This PR removes the following charts (which were marked as deprecated in 3.0 though in actuality were obsolete):

- `TableViz`
- `BigNumberViz`
- `BigNumberTotalViz`

which will ensure that the logic in https://github.com/apache/superset/pull/24671 will be correct.

Technically (as illustrated by the augmented tests) these visualization types could be invoked either using the legacy or non-legacy RESTful API endpoint, though in actuality (per the frontend logic) they would only ever be invoked under the later.

### TESTING INSTRUCTIONS

I systematically went through all the visualization types in `superset/viz.py` (all NVD3 and DeckGL charts were bucketed together) and inspected which RESTful API endpoint was being invoked, i.e., any chart using the RESTful `/superset/explore_json` endpoint (or similar) remained.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/integration_tests/cache_tests.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def test_no_data_cache(self):
         app.config["DATA_CACHE_CONFIG"] = {"CACHE_TYPE": "NullCache"}
         cache_manager.init_app(app)
 
-        slc = self.get_slice("Girls", db.session)
+        slc = self.get_slice("Top 10 Girl Name Share", db.session)
```
Comment (john-bodley) [inline tests/integration_tests/cache_tests.py:49]: Switching out for a legacy chart which invokes the `/superset/explore_json` RESTful API endpoint.

File: tests/integration_tests/charts/api_tests.py [hunk 2]
```diff
@@ -1715,7 +1715,7 @@ def test_gets_owned_created_favorited_by_me_filter(self):
     )
     def test_warm_up_cache(self):
         self.login()
-        slc = self.get_slice("Girls", db.session)
+        slc = self.get_slice("Top 10 Girl Name Share", db.session)
```
Comment (john-bodley) [inline tests/integration_tests/charts/api_tests.py:1718]: Currently caching only works with legacy charts and removal of the `TableViz` now treats the `Girls` chart as a non-legacy chart. One could argue this is a regression—in terms of fewer charts being eligible for caching—but the problem will be rectified in https://github.com/apache/superset/pull/24671.

File: tests/integration_tests/cache_tests.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def test_slice_data_cache(self):
         }
         cache_manager.init_app(app)
 
-        slc = self.get_slice("Boys", db.session)
+        slc = self.get_slice("Top 10 Girl Name Share", db.session)
```
Comment (john-bodley) [inline tests/integration_tests/cache_tests.py:76]: See previous comment.

File: tests/integration_tests/core_tests.py [hunk 4]
```diff
@@ -391,7 +393,7 @@ def test_databaseview_edit(self, username="admin"):
     )
     def test_warm_up_cache(self):
         self.login()
-        slc = self.get_slice("Girls", db.session)
+        slc = self.get_slice("Top 10 Girl Name Share", db.session)
```
Comment (john-bodley) [inline tests/integration_tests/core_tests.py:396]: See previous comment.

File: tests/integration_tests/core_tests.py [hunk 5]
```diff
@@ -418,10 +420,10 @@ def test_cache_logging(self):
         self.login("admin")
         store_cache_keys = app.config["STORE_CACHE_KEYS_IN_METADATA_DB"]
         app.config["STORE_CACHE_KEYS_IN_METADATA_DB"] = True
-        girls_slice = self.get_slice("Girls", db.session)
-        self.get_json_resp(f"/superset/warm_up_cache?slice_id={girls_slice.id}")
+        slc = self.get_slice("Top 10 Girl Name Share", db.session)
```
Comment (john-bodley) [inline tests/integration_tests/core_tests.py:423]: See previous comment.

File: tests/integration_tests/security_tests.py [hunk 6]
```diff
@@ -1680,7 +1680,7 @@ def test_raise_for_access_table(self, mock_can_access):
     def test_raise_for_access_viz(
         self, mock_can_access_schema, mock_can_access, mock_is_owner
     ):
-        test_viz = viz.TableViz(self.get_datasource_mock(), form_data={})
+        test_viz = viz.TimeTableViz(self.get_datasource_mock(), form_data={})
```
Comment (john-bodley) [inline tests/integration_tests/security_tests.py:1683]: Any legacy viz type works.

File: tests/integration_tests/viz_tests.py [hunk 7]
```diff
@@ -45,7 +45,7 @@ def test_constructor_exception_no_datasource(self):
             viz.BaseViz(datasource, form_data)
 
     def test_process_metrics(self):
-        # test TableViz metrics in correct order
+        # test TimeTableViz metrics in correct order
```
Comment (john-bodley) [inline tests/integration_tests/viz_tests.py:48]: Any legacy viz type should work.

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thanks for the cleanup.

**Comment by john-bodley** [comment]
Regrettably closing in favor of https://github.com/apache/superset/pull/24692.

**Comment by john-bodley** [comment]
Closing in favor of https://github.com/apache/superset/pull/24692.

**Comment by john-bodley** [comment]
Closing in favor of https://github.com/apache/superset/pull/24694.

**Pull request closed without merging**